app/src/main/java/com/example/scoutconcordia/Activities/MapsActivity.java

```diff
@@ -715,85 +723,126 @@ public void onGetDirectionsClick(View v){
 
     public void getDirections()
     {
+        String startingPointType = "class"; // can be type class or building
+        String destinationType = "class";  // can be type class or building
+
+        for (int i = 0; i < restaurantMarkers.size(); i++)  // Check if either the destination or the starting point is a restaurant
+        {
+            if (restaurantMarkers.get(i).getTitle().equals(startingPoint))
+            {
+                startingPointType = "building";
+            }
+            else if (restaurantMarkers.get(i).getTitle().equals(destination))
+            {
+                destinationType = "building";
+            }
+        }
+
+        if (startingPoint.length() > 8 && startingPoint.substring(startingPoint.length() - 8).equals("Building"))  // check if the starting point is a building
```

Comment:
Many of our code smells detected by SonarQube are due to us using literal strings instead of constants, can we declare "Building" as a constant and use the variable instead?

For example: 
```
private static final String Building = "Building";

Reply:
Constants should be in all caps.

`private static final String BUILDING = "Building";`

Reply:
We can make this modification when we refactor the whole code, what do you think @jaychenblue